File: Abel_Elias/src/classes/expressions/BooleanLiteral.java
package classes.expressions;

import classes.CodeBlock;

public class BooleanLiteral extends Literal {

    private final Boolean value;

    public BooleanLiteral(CodeBlock code, Boolean value) {
        super(code);
        this.value = value;
    }

    public BooleanLiteral(Boolean value) {
        super(new CodeBlock(DUMMY_LINE, DUMMY_LINE));
        this.value = value;
    }

    public Boolean getValue() {
        return this.value;
    }


    @Override
    public <T> T accept(ExpressionVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: Carly_Michael/jectacular-ql/src/app/domain/ast/ql/expressions/literals/boolean-literal.ts
import {Location} from '../../../location';
import {ExpressionType} from '../expression-type';
import {ExpressionVisitor} from '../../visitors/expression-visitor';
import {Literal} from './literal';

export class BooleanLiteral extends Literal {
  constructor(public readonly value: boolean, readonly location: Location) {
    super(location);
  }

  getValue(): any {
    return this.value;
  }

  getType(): ExpressionType {
    return ExpressionType.BOOLEAN;
  }

  accept<T>(visitor: ExpressionVisitor<T>): T {
    return visitor.visitBooleanLiteral(this);
  }

  // boolean ops

  equals(literal: Literal): Literal {
    return literal.equalsBoolean(this);
  }

  equalsBoolean(literal: BooleanLiteral): Literal {
    return new BooleanLiteral(this.value === literal.value, this.location);
  }

  notEquals(literal: Literal): Literal {
    return literal.notEqualsBoolean(this);
  }

  notEqualsBoolean(literal: BooleanLiteral): Literal {
    return new BooleanLiteral(this.value !== literal.value, this.location);
  }

  or(literal: Literal): Literal {
    return literal.orBoolean(this);
  }

  orBoolean(literal: BooleanLiteral): Literal {
    return new BooleanLiteral(this.value || literal.value, this.location);
  }

  and(literal: Literal): Literal {
    return literal.andBoolean(this);
  }

  andBoolean(literal: BooleanLiteral): Literal {
    return new BooleanLiteral(this.value && literal.value, this.location);
  }

  negation(): Literal {
    return new BooleanLiteral(!this.value, this.location);
  }
}
